packages/ts-exporter/src/test/java/org/uberfire/jsbridge/tsexporter/dependency/DependencyGraphTest.java

```diff
@@ -0,0 +1,219 @@
+    public void testSimpleGraphVertices() {
+        final DependencyGraph graph = new DependencyGraph(Stream.empty(), NO_DECORATORS);
+        graph.add(element(A0.class));
+        assertEquals(1, graph.vertices().size());
+        graph.add(element(A1.class));
+        assertEquals(2, graph.vertices().size());
+        graph.add(element(A2.class));
+        assertEquals(3, graph.vertices().size());
+
+        assertEquals(list(), simpleNames(graph.vertex(element(A0.class)).dependencies.keySet()));
+        assertEquals(list("A0"), simpleNames(graph.vertex(element(A1.class)).dependencies.keySet()));
+        assertEquals(list("A1"), simpleNames(graph.vertex(element(A2.class)).dependencies.keySet()));
+
+        assertEquals(list("A1"), simpleNames(graph.vertex(element(A0.class)).dependents.keySet()));
+        assertEquals(list("A2"), simpleNames(graph.vertex(element(A1.class)).dependents.keySet()));
+        assertEquals(list(), simpleNames(graph.vertex(element(A2.class)).dependents.keySet()));
+    }
+
+    @Test
+    public void testGraphVerticesComplex() {
+        final DependencyGraph graph = new DependencyGraph(Stream.empty(), NO_DECORATORS);
+        graph.add(element(A2B1.class));
+        assertEquals(4, graph.vertices().size());
+        graph.add(element(A1.class));
+        assertEquals(4, graph.vertices().size());
+        graph.add(element(B0.class));
+        assertEquals(4, graph.vertices().size());
+        graph.add(element(A1B1.class));
+        assertEquals(5, graph.vertices().size());
+        graph.add(element(A0.class));
+        assertEquals(5, graph.vertices().size());
+        graph.add(element(A2.class));
+        assertEquals(6, graph.vertices().size());
+
+        assertEquals(list(), simpleNames(graph.vertex(element(A0.class)).dependencies.keySet()));
+        assertEquals(list("A0"), simpleNames(graph.vertex(element(A1.class)).dependencies.keySet()));
+        assertEquals(list("A0", "B0"), simpleNames(graph.vertex(element(A1B1.class)).dependencies.keySet()));
+        assertEquals(list("A1"), simpleNames(graph.vertex(element(A2.class)).dependencies.keySet()));
+        assertEquals(list("A1", "B0"), simpleNames(graph.vertex(element(A2B1.class)).dependencies.keySet()));
+        assertEquals(list(), simpleNames(graph.vertex(element(B0.class)).dependencies.keySet()));
+
+        assertEquals(list("A1", "A1B1"), simpleNames(graph.vertex(element(A0.class)).dependents.keySet()));
+        assertEquals(list("A2", "A2B1"), simpleNames(graph.vertex(element(A1.class)).dependents.keySet()));
+        assertEquals(list(), simpleNames(graph.vertex(element(A1B1.class)).dependents.keySet()));
+        assertEquals(list(), simpleNames(graph.vertex(element(A2.class)).dependents.keySet()));
+        assertEquals(list(), simpleNames(graph.vertex(element(A2B1.class)).dependents.keySet()));
+        assertEquals(list("A1B1", "A2B1"), simpleNames(graph.vertex(element(B0.class)).dependents.keySet()));
+    }
+
+    class c0 {
```

Comment:
curious: are these classes starting with lower-case letters for testing purposes?
If not I think it would be good to follow regular naming conventions for tests too